### PR TITLE
Codec Adapter: Use the passed input buffers

### PR DESCRIPTION
**Pull request by ranj063**
Use the passed input buffers and move the initialization call within the first process callback for each codec.

### CODE REVIEW

**Review by lgirdwood — COMMENTED** [review]
Looks like we get an new memcpy() during process() - is this temporary until the other changes are in place ?

File: src/audio/codec_adapter/codec_adapter.c [hunk 1]
```diff
@@ -395,9 +395,6 @@ int codec_adapter_copy(struct comp_dev *dev)
 		return -EINVAL;
 	}
```
Comment (dbaluta) [inline src/audio/codec_adapter/codec_adapter.c:397]: @ranj063 for this reason this patch breaks compress play on i.MX. WIll investigate it and come back with more info.

**Comment by ranj063** [comment]
> Looks like we get an new memcpy() during process() - is this temporary until the other changes are in place ?

@lgirdwood yes, eventually after I switch all the codecs to use the input/output buffers, the memcpy will be removed as well

File: src/audio/codec_adapter/codec_adapter.c (hunk 1 above)
Comment (ranj063) [reply src/audio/codec_adapter/codec_adapter.c:397]: @dbaluta patch 2 alone will break the cadence implementation. Can you please let me know what error you're seeing with patch 1, 2 & 3?

**Review by cujomalainey — APPROVED** [review]

**Comment by miRoox** [comment]
SOFCI TEST

**Comment by dbaluta** [comment]
@ranj063 i.MX compress works fine after your last patch.

**Review by dbaluta — APPROVED** [review]

**Pull request merged**